### ISSUE TITLE
VM stalls on deferred load of package URI when it has no way to resolve packages.

### ISSUE THREAD

**Issue by lrhn** [dart-lang/sdk#26482]
Example:
```dart
import "package:foo/foo.dart" deferred as foo;
main() async {
  await foo.loadLibrary().timeout(new Duration(seconds: 10)).catchError(print);
}
```
If you run this in a VM with no package root or package config, the timeout triggers. Without the timeout, the program just exits because the future will never be completed.

If you run it with a package root or config, even one with no "foo" package in it, the `loadLibrary` call just fails with an error. 

The stalling behavior is surprising and can easily make your program exit when you await the result.

**Comment by mit-mit** [comment]
Moving to @a-siva for assignment

**Comment by mhausner** [comment]
I think this is a duplicate of bug #23797.

I don't observe a difference in behavior if a package-root is given on the command line or not. In both cases, the timeout fires and is printed by the catchError() handler. If no timeout is given, the VM terminates and the loadLibrary() future is never completed.

With timeout:
```
import "package:foo/foo.dart" deferred as foo;
main() async {
  await foo.loadLibrary().timeout(new Duration(seconds: 2)).catchError(print);
  print("after await");
}

$ $DEBUG_DIR/dart ~/tmp/p.dart
TimeoutException after 0:00:02.000000: Future not completed
after await

$ $DEBUG_DIR/dart --package-root=$DEBUG_DIR/packages/ ~/tmp/p.dart
TimeoutException after 0:00:02.000000: Future not completed
after await
```

Without the timeout:

```
import "package:foo/foo.dart" deferred as foo;

main() async {
  await foo.loadLibrary().catchError(print);
  print("after await");
}


$DEBUG_DIR/dart ~/tmp/p.dart
$

$ $DEBUG_DIR/dart --package-root=$DEBUG_DIR/packages/ ~/tmp/p.dart
$
```

**Comment by floitschG** [comment]
Why is the `loadLibrary` never completed?

**Comment by lrhn** [comment]
Florian is asking the correct question.

If the VM has a way to resolve packages, but the requested file does not exist, the future completes with an error.

If the VM has no way to resolve packages, the future does not complete. 

It should complete in both cases, with an error. Not completing isn't useful, and you have to be defensive and add a timeout to get the expected behavior.

So, this is not a duplicate of #23797 - the problem isn't the behavior when the future doesn't complete, but that it doesn't complete.

**Comment by mhausner** [comment]
I agree. This is a recent regression, as it turns out. We are looking at it now.

**Comment by mhausner** [comment]
@johnmccutchan fixed this in 3b2e14ab56d2e5e9dbf660270153402e6689bd1e. Thank you!
